src/test/prediction_service_test.hpp

```diff
@@ -0,0 +1,240 @@
+//*****************************************************************************
+// Copyright 2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#pragma once
+
+#include <cstdio>
+#include <cstring>
+#include <filesystem>
+#include <fstream>
+#include <future>
+#include <memory>
+#include <string>
+#include <thread>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <inference_engine.hpp>
+#include <stdlib.h>
+
+#include "../executinstreamidguard.hpp"
+#include "../modelinstance.hpp"
+#include "../prediction_service_utils.hpp"
+#include "test_utils.hpp"
+
+using testing::Each;
+using testing::Eq;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnarrowing"

```

Comment:
Why you need those? I think we needed those only for includes from TensorflowServing.

Reply:
Some tests also needed exceptions in warnings.